config/routes.rb

```diff
@@ -32,12 +32,17 @@
   end
 
   resources :collections do
+    collection do
+      post :create_ajax
+    end
+
     member do
       get :download_all
       get :view_shared # ???
 
       patch :remove_all
       patch :remove_task
+      patch :remove_task_ajax
```

Comment:
You can restrict routes as follows, assuming they shall only be available for AJAX calls:

```suggestion
      patch :remove_task_ajax, constraints: ->(r) { r.xhr? }
```

Reply:
This is a leftover now, isn't it?